statsd/statsdtest/recorder.go

```diff
@@ -0,0 +1,85 @@
+package recorder
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/falun/go-statsd-client/statsd"
+)
+
+// RecordingSender implements statsd.Sender but parses individual Stats into a
+// buffer that can be later inspected instead of sending to some server. It
+// should constructed with NewRecordingSender().
+type RecordingSender struct {
+	m      sync.Mutex
+	buffer Stats
+	closed bool
+}
+
+var _ statsd.Sender = &RecordingSender{}
```

Comment:
Oh, one weird bit about that is that since it's following the same model the Test itself won't actually test anything explicitly but will still be acting as a compile time check (which is why typically i put them in the definition).

I can push the update (compilation failure in test code) or leave it as is, let me know what feels better. (I guess you could also argue to remove it completely but I like having the assurances around that your types fill the the interfaces you think they do)